src/main/steps/common/confirm-contact-details/checkanswers/ConfirmContactDetailsPostController.ts

```diff
@@ -84,24 +87,28 @@ export const saveAndRedirectContactDetailsAndPreference = async (
       },
     });
 
-    try {
-      req.session.userCase = await client.updateCaseData(
-        userCase.id,
-        partyDetails,
-        partyType,
-        userCase.caseTypeOfApplication as CaseType,
-        req.session.applicationSettings?.navFromContactPreferences
-          ? CaseEvent.CONTACT_PREFERENCE
-          : CaseEvent.CONFIRM_YOUR_DETAILS
-      );
-      mapDataInSession(req.session.userCase, user.id);
-      req.session.userCase.citizenUserAddressText = (await setAddressFields(req)).citizenUserAddressText;
-      req.session.save(() => {
-        const redirectUrl = getRedirectUrl(partyType, req);
-        res.redirect(redirectUrl);
-      });
-    } catch (error) {
-      throw new Error('ConfirmContactDetailsPostController - Case could not be updated.');
+    if (userCase?.isCitizenLivingInRefuge === YesOrNo.YES && _.isEmpty(userCase?.refugeDocument)) {
```

Comment:
have a method called isMandatoryFieldsFilled and then allow to submit if the return value is true.